src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -174,7 +174,12 @@ private async ValueTask ReloadProjectsAsync(ImmutableSegmentedList<ProjectToLoad
         }
     }
 
-    protected sealed record RemoteProjectLoadResult(RemoteProjectFile ProjectFile, ProjectSystemProjectFactory ProjectFactory, bool IsMiscellaneousFile, BuildHostProcessKind Preferred, BuildHostProcessKind Actual);
+    /// <param name="HasLoadErrors">
+    /// Whether there were errors in the process of loading the project file itself.
+    /// When this is true, we may still want to proceed with design-time build in order to provide partial information about the project.
```

Comment:
The semantics described here are a bit reversed of what I'd imagine we would do -- I would have expected 'true' means 'we know there are already errors, no reason to go further'.

Is the easier approach here just to return a set of diagnostics that are concatenated onto the existing things that get reported?